connect-button/src/main/java/com/ifttt/connect/ui/AnalyticsApiHelper.java

```diff
@@ -15,34 +16,38 @@
 
     private final EventsApi eventsApi;
 
-    private AnalyticsApiHelper() {
+    private AnalyticsApiHelper(Context context) {
+        Interceptor interceptor = (chain -> chain.proceed(chain.request()
+                .newBuilder()
+                .addHeader("IFTTT-SDK-Version", BuildConfig.VERSION_NAME)
+                .addHeader("IFTTT-SDK-Platform", "android")
+                .addHeader("IFTTT-SDK-Anonymous-Id", AnalyticsPreferences.getAnonymousId(context))
```

Comment:
This helper actually only needs a string anonymous id as dependency, not `context`, right? It just happens to be that the string is coming from SharedPreferences.

If so, we should change this constructor to take in a string as param, and let the static method or the caller of the static method figure out where to get the string.